proposal/adkevin3307/proposal.md

```diff
@@ -0,0 +1,124 @@
+# Proposal for Body IsoSurface
+
+## Basic information
+
+[GitHub](https://github.com/adkevin3307/body_isosurface)
+
+Plot the isosurface with the intensity of body tissue.
+
+## Problem to solve
+
+When humans taking Magnetic Resonance Imaging (MRI) or Computed Tomography (CT), the computer will 
+get the intensity of body tissues. We need some algorithms to visualize the intensity data, and 
+isosurface is one of them. By plotting the same value of intensity, we can easily recognize specific 
+tissue in the body, and also easily find the abnormal place. In summary, I want to develop a method to 
+calculate an isosurface with C++, and maybe provide 3D rendering methods for output triangles.
+
+### Algorithm and Mathematics
+
+- Marching Cubes
+    - Calculate intersections for each cube (8 voxel per cube)
+    - Decide the way that the plane in the cube can be separated to triangles
+    - [Marching cubes (wiki)](https://en.wikipedia.org/wiki/Marching_cubes)
+
+### Target
+
+- Well-Defined Structure
+- CI Procedure
+- Computation Speed
+- 3D Rendering (maybe)
+    - Camera
+    - Shadow
+    - Controllable
+
+## Perspective users
+
+All people who is interested in body tissues, special for doctors who need to read MRI and CT results.
+
+## System architecture
+
+![](body_isosurface.png)
+
+The data flow is shown above. Each data has inf file and raw file, inf file records information for data, 
+like `endian`, `data type`, `volume size`, and so on, and the intensity values are placed in raw file. 
+First, read data with `Volume` class and calculate each voxel's normal. After the `Volume` class calculate 
+finish, pass the result to the `IsoSurface` class. The `Volume` class need to decide iso value automatically 
+if the user does not define it. When the procedures mentioned above are done, the `IsoSurface` class will 
+calculate and separate the isosurface with multiple triangles that follows Marching Cubes method. Then the 
+user can get and use triangles and normals for the data.
+
+## API description
+
+- Volume class
```

Comment:
Descriptions like this may go directly into code and you may provide URL to your github.  But this format works for the course, too.